ml-proto/src/script.ml

```diff
@@ -71,6 +75,42 @@ let run_command cmd =
       Error.error cmd.at "assertion failed"
     end
 
+  | AssertHeapEq (offset, expected) ->
```

Comment:
Instead of checking each character individually, how about having a loop that reads the actual bytes into a `bytes` buffer of suitable length, and then just compare `Bytes.to_string buf = expected`. That way, you don't need to implement a second loop just for error printing.